src/zarr/storage/_register_adapters.py

```diff
@@ -0,0 +1,50 @@
+"""
+Auto-registration of built-in store adapters.
+
+This module ensures that built-in store adapters are registered
+when zarr-python is imported, providing ZEP 8 URL syntax support
+out of the box.
+"""
+
+from zarr.registry import register_store_adapter
+
+
+def register_builtin_adapters() -> None:
+    """Register all built-in store adapters."""
+    # Import all the adapter classes
+    # Register all adapters
+    from typing import TYPE_CHECKING
+
+    from zarr.storage._builtin_adapters import (
+        FileSystemAdapter,
+        GSAdapter,
+        HttpsAdapter,
+        LoggingAdapter,
+        MemoryAdapter,
+        S3Adapter,
+        S3HttpAdapter,
+        S3HttpsAdapter,
+        ZipAdapter,
```

Comment:
We should add `LatencyStore` to this list with some sensible defaults it would be super helpful for testing